### PR TITLE
[ spi_host ] Transition to D1

### PR DESCRIPTION
**Pull request by martin-lueker**
This IP is largely ready for transition to D1.  Please reach out to me if you need a review.

Signed-off-by: Martin Lueker-Boden <martin.lueker-boden@wdc.com>

### CODE REVIEW

**Comment by martin-lueker** [comment]
FYI @mateenfaisal

**Comment by mateenfaisal** [comment]
I believe @eunchan is the primary reviewer for SPI Host IP. He works on SPID. Right, @dsandrs?

File: hw/ip/spi_host/doc/checklist.md
Comment (imphil) [inline hw/ip/spi_host/doc/checklist.md:24]: @msfschaffner Can you have a look at that? If we really want to go to a macro we should plan for it soon I guess. (But if I remember the discussions right the cut-off point for SRAM is rather late, so probably FFs it is here?)

File: hw/ip/spi_host/doc/checklist.md
Comment (imphil) [inline hw/ip/spi_host/doc/checklist.md:26]: The `passthrough_o` is missing an known assert.

File: hw/ip/spi_host/doc/checklist.md
Comment (imphil) [inline hw/ip/spi_host/doc/checklist.md:19]: For the purpose of D1, the documentation is Done, and you can mark is as such ("Mostly Done" wouldn't really get you D1 approval :smile:)

Also, can you change the doc link to go to `.` (the current directory?) I have been cleaning up most such links to make IPs easier to move around between directories, but seems like I missed some.

```suggestion
Documentation | [SPEC_COMPLETE][]              | Mostly Done | Programming model still needs work, noted in spec [SPI_HOST Design Spec]({{<relref "." >}})
```

**Review by imphil — COMMENTED** [review]
Almost an approval from my side :-)

File: hw/ip/spi_host/doc/checklist.md
Comment (msfschaffner) [reply hw/ip/spi_host/doc/checklist.md:24]: That will have a big floorplan impact at this point, and would certainly cause issues on the Nuvoton side.
What configurations are we talking about?

File: hw/ip/spi_host/doc/checklist.md
Comment (martin-lueker) [reply hw/ip/spi_host/doc/checklist.md:19]: Fair enough. :)  I've got the Doc open on my desktop, should be done by the weekend.

Assuming the Doc review is uncontroversial (it really just clarifies the existing implementation details), it is quite possible this IP will stay very briefly in D2.

I'll update the PR # for the doc, when I get a new one.

File: hw/ip/spi_host/doc/checklist.md
Comment (martin-lueker) [reply hw/ip/spi_host/doc/checklist.md:26]: Good catch, thanks.

File: hw/ip/spi_host/doc/checklist.md
Comment (martin-lueker) [reply hw/ip/spi_host/doc/checklist.md:24]: There are two 32w x64deep async fifos.   In addition, one of these has a small sync fifo after the async.  IIRC and looking at conversations such as in https://github.com/lowRISC/opentitan/issues/5112 this is near the crossover right?

File: hw/ip/spi_host/doc/checklist.md
Comment (martin-lueker) [reply hw/ip/spi_host/doc/checklist.md:26]: added.

File: hw/ip/spi_host/doc/checklist.md
Comment (msfschaffner) [reply hw/ip/spi_host/doc/checklist.md:24]: Yes this exactly around the crossover point (SRAMs could potentially be slightly denser). 
So given the floorplan impact I would refrain from making these SRAM macros now.

Comment (msfschaffner) [reply hw/ip/spi_host/doc/checklist.md:24]: Can you remove the comment?

**Review by msfschaffner — APPROVED** [review]
LGTM, modulo the comment on SRAM.

File: hw/ip/spi_host/doc/checklist.md
Comment (martin-lueker) [reply hw/ip/spi_host/doc/checklist.md:24]: Thanks Michael, sure will do.

**Comment by martin-lueker** [comment]
- The functional documentation is now complete (minus implementation details) and in #7135
- The note about SRAMs has been removed.
- Some of the D2 details have been checked off (I'll be coming back around late next week to check off more)

**Comment by imphil** [comment]
We can't merge this before we have a "Done" in the spec column. So #7135 needs to go in first, but beyond that there's nothing blocking from my side.

File: hw/ip/spi_host/doc/checklist.md
Comment (imphil) [inline hw/ip/spi_host/doc/checklist.md:19]: Let's mark this "DONE" as soon as https://github.com/lowRISC/opentitan/pull/7135 is in. And please add back a link (relative links preferred)

```
[SPI_HOST Design Spec]({{<relref "." >}})
```

**Review by imphil — APPROVED** [review]
LGTM with the updated SPEC_COMPLETE column. Thanks Martin!

File: hw/ip/spi_host/doc/checklist.md
Comment (martin-lueker) [reply hw/ip/spi_host/doc/checklist.md]: Great.  Thanks Philipp and all!

**Pull request merged**
